pkg/controllers/cspi-controller/handler.go

```diff
@@ -526,6 +527,31 @@ func (c *CStorPoolInstanceController) reconcileVersion(cspi *cstor.CStorPoolInst
 	return cspi, nil
 }
 
+// markCSPIStatusToOffline will fetch all the CSPI resources present
+// in etcd and mark it's own CSPI.Status to Offline
+func (c *CStorPoolInstanceController) markCSPIStatusToOffline() {
+	cspiList, err := c.clientset.CstorV1().CStorPoolInstances("").List(metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Failed to fetch CSPI list error: %v", err)
+	}
+	// Fetch CSPI UUID of this pool-manager
+	cspiID := os.Getenv(OpenEBSIOCSPIID)
+	for _, cspi := range cspiList.Items {
+		if string(cspi.GetUID()) == cspiID {
+			cspi.Status.Phase = cstor.CStorPoolStatusOffline
```

Comment:
This will cause a problem if the pool is not created and if the pod is restarted. Need to add a check over here.

Reply:
done